### ISSUE TITLE
TypeError: Bad argument when starting wallaby

### ISSUE THREAD

**Issue by kellyethridge** [wallabyjs/public#993]
### Issue description or question
Wallaby will fail to start then stop and report a Type Error. Looking in the log, it reports:

> 2017-02-06 08:27:54Z [1] Error Wallaby.Menu.StartRunnerCommand 0: Error occurred during selecting configuration project item.
> System.ArgumentException: The parameter is incorrect. (Exception from HRESULT: 0x80070057 (E_INVALIDARG))
>    at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
>    at Microsoft.VisualStudio.PlatformUI.UIHierarchyDTEHelpers.NodeFromName(IEnumerable`1 childNodes, String name, Boolean ignoreCase)
>    at Microsoft.VisualStudio.PlatformUI.UIHierarchyMarshaler.UIHierarchy.GetItem(String Names)
>    at Microsoft.VisualStudio.PlatformUI.UIHierarchyMarshaler.<>c__DisplayClass6_0.<GetItem>b__0()
>    at Microsoft.VisualStudio.Shell.ThreadHelper.Invoke[TResult](Func`1 method)
>    at Microsoft.VisualStudio.PlatformUI.UIHierarchyMarshaler.GetItem(String Names)
>    at ⁫‏⁭⁯‬⁬‎⁯​‭‮​‎⁯‮‏‬⁬⁬‫‍⁮⁮⁮‎⁮⁫⁮⁪​⁫⁮‪‎‮‬‪‎‮‮.Oi5bOzJh!A!1r=m^XP~YpcHF.MoveNext()
> 

I'll go round-n-round uninstalling and re-installing "electron" or try using Phantomjs and once in a while it will work, but then will just fail the next time I reload the solution. Maybe my config is just completely wrong.

### Wallaby.js configuration file
```javascript
var Wallabify = require("wallabify");
var WallabyPostprocessor = Wallabify({
    entryPatterns: [
      "Web/Content/Scripts/tests/jsnlogTestConfigure.js",
      "Web/Content/Scripts/tests/**/test.*.js*"
    ]
});
var System;
var Promise;

module.exports = function (wallaby) {
    return {
        files: [
          { pattern: "node_modules/babel-polyfill/browser.js", instrument: false },
          { pattern: "jspm_packages/system.js", instrument: false },
          { pattern: "config.js", instrument: false },
          { pattern: "app/**/*.ts", load: false, instrument: true },
          { pattern: "tests/**/fake*.ts", load: false, instrument: false }
        ],
        tests: [
            { pattern: "tests/**/*Tests.ts", load: false },
            { pattern: "tests/**/*-tests.ts", load: false }
        ],
        compilers: {
            "**/*.ts": wallaby.compilers.typeScript({
                module: "amd",
                target: "es5",
                emitDecoratorMetadata: true,
                experimentalDecorators: true,
                noImplicitAny: false
                // TypeScript compiler specific options
                // See interface CompilerOptions in
                // https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts
            }),
            postprocessor: WallabyPostprocessor
            //preprocessors: [{
            //    '**/*.js': file => require("babel-core")
            //        .transform(file.content, {
            //            sourceMap: true,
            //            compact: false,
            //            presets: ["es2015"]
            //        })

            //}]
        },
        delays: {
            edit: 400,
            run: 150
        },
        env: {
            kind: "electron"
        },
        testFramework: "jasmine",

        middleware: (app, express) => {
            app.use("/jspm_packages", express.static(require("path").join(__dirname, "jspm_packages")));
        },

        bootstrap: function (w) {
            w.delayStart();

            System.config({
                paths: {
                    '*': "*.js",
                    "github:*": "/jspm_packages/github/*",
                    "npm:*": "/jspm_packages/npm/*"
                }
            });

            var promises = [];

            for (var i = 0, len = w.tests.length; i < len; i++) {
                promises.push(System["import"](w.tests[i].replace(/\.js$/, "")));
            }

            Promise.all(promises).then(function () {
                w.start();
            });
        },

        debug: false
    };
};
```
### tsconfig.json
```javascript
{  
  "compileOnSave": true,
  "compilerOptions": {
    "rootDir": "/",
    "sourceMap": true,
    "target": "es5",
    "module": "amd",
    "declaration": false,
    "noImplicitAny": false,
    "noResolve": false,
    "removeComments": true,
    "noLib": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true
  },
  "filesGlob": [
    "./app/**/*.ts",
    "./typings/index.d.ts",
    "./scripts/typings/**/*.d.ts"
  ],
  "exclude": [
    "node_modules",
    "jspm_packages"
  ]
}
```

### Code editor or IDE name and version

Visual Studio 2015 (Community)

### OS name and version
Windows 10

**Comment by ArtemGovorov** [comment]
I don't think there's any issue with your config file, it looks like the issue with the extension.

What is the installed extension version in [the Extension and Updates](https://wallabyjs.com/docs/intro/install.html#visual-studio)?

Could you please also list other extensions that you have installed?

**Comment by ArtemGovorov** [comment]
Please also paste the full output from [the Wallaby Console](https://wallabyjs.com/assets/img/console.png).

**Comment by ArtemGovorov** [comment]
Actually, your config also contains a couple of things that don't make sense to me. Your entry patterns, that you pass to `Wallabify` are supposed to be some files from your `files` list and your `tests`, but instead, they seem to be referencing some folder that is not even in the `files` list (`Web/Content/Scripts`):
```javascript
      "Web/Content/Scripts/tests/jsnlogTestConfigure.js",
      "Web/Content/Scripts/tests/**/test.*.js*"
```

**Comment by ArtemGovorov** [comment]
I'm not sure how your folder structure looks like, but if it looks like I think it looks like, then you may try this:

```javascript
var WallabyPostprocessor = Wallabify({
    entryPatterns: [
      "tests/jsnlogTestConfigure.js",
      "tests/**/*Tests.js",
      "tests/**/*-tests.js"
    ]
});
```

also, it doesn't look like the `tests/jsnlogTestConfigure` file is listed anywhere in any patterns of the `files` list, so you'll need to add it there as well.

If the suggestions above don't help, please create and share a sample repo (reflecting your flder structure and with your config files, and a dummy test/source file), where the issue can be reproduced, happy to have a look.

**Comment by kellyethridge** [comment]
The config file was a template I found long ago when trying to make TypeScript work with Wallaby. It clearly needs cleaning up.

OK, I found the issue. Nothing to do with Wallaby and everything to do with my naive npm package management skills. I had installed 'electron' (not the JS engine) once and did not uninstall it correctly. Bower seemed to keep re-installing it and it somehow interfered with 'electron-prebuilt' I'm guessing.

Again, thank you for your support and great product!
